### PR TITLE
Add motion magic control mode

### PR DESCRIPTION
**Pull request by Twinters007**
I think I got everything needed.

### CODE REVIEW

**Comment by virtuald** [comment]
What's with all the whitespace/other random corrections? Can you separate those into a separate commit, that's just annoying.

**Comment by Twinters007** [comment]
It's eclipse. I have autopep on normally and didn't turn it off. I'll try and get rid of them.

**Comment by FRC2539** [comment]
It seems like you'd need to add a `MotionMagic` constant to the `CANTalon.ControlMode` class, and there there might need to be logic added to the `CANTalon.set()` method as well.

Sorry if we shouldn't have jumped in on this ticket yet, but we're really excited to see this mode in action.

**Comment by virtuald** [comment]
@FRC2539 Feel free to comment/review on any PRs that you might see anywhere in RobotPy, I appreciate all the input I can get! :)

And you're totally right, it should be added to https://github.com/Twinters007/robotpy-ctre/blob/324de457ba56f976360d0eaf64db8ec597d12a18/ctre/cantalon.py#L164 ... there might be logic required for `set()` too.

**Comment by Twinters007** [comment]
Darn. That one got away when I reset and recommitted.

File: ctre/cantalon.py [hunk 1]
```diff
@@ -442,6 +443,8 @@ def set(self, outputValue, syncGroup=0):
             self.handle.SetDemand(int(mA))
         elif self.controlMode == self.ControlMode.MotionProfile:
             self.handle.SetDemand(outputValue)
+        elif self.controlMode == self.ControlMode.MotionMagic:
+            self.handle.SetDemand(self.scaleRotationsToNativeUnits(self.feedbackDevice, -outputValue if self.isInverted else outputValue))
```
Comment (virtuald) [inline ctre/cantalon.py]: I think scale starts with a `_`... see line 440? Also, why scaleRotations?

Comment (Twinters007) [inline ctre/cantalon.py]: idk. That's just what they had in the jar

**Comment by virtuald** [comment]
Thanks!

**Pull request merged**